enhancements/ingress/set-delete-http-headers.md

```diff
@@ -1001,17 +1000,45 @@ This route can be annotated with `haproxy.router.openshift.io/disable_cookies=tr
-  The purpose of this EP is to define custom headers. So, there are two reasons we need such override:
-  1. A route owner shall have the choice to define what headers he wants as per his application requirements(as they might be varying) 
-  than the ones admin has set cluster-wide.
-  2. The IngressController and Route might be owned by the same person who wants to specify some global configuration
-  and then override that global configuration on a small subset of routes. 
+#### How do Route headers interact with IngressController headers of the same value?
+  Answer:  It depends on if they are response headers or request headers.
+  
+ In case of HTTP response headers, the headers set in spec.httpHeaders.actions of IngressController will override
+ the headers with same name in Route's spec.httpHeaders.actions or Route annotation.
+
+ In case of HTTP request headers, the headers set in Route using spec.httpHeaders.actions or annotation will override the 
+ headers with the same name present in the IngressController's spec.httpHeaders.actions.
+
+ This is because of the way HAProxy is designed to work. We cannot change the processing order; we're constrained by HAProxy's design: 
+ the frontend receives the request and dispatches it to the backend, and then the frontend gets the 
+ response from the backend.
+
+ In case of HTTP response, HAProxy overrides the header value of the same header name in the backend with
+ the value of the header defined in the frontend. So, [the backend rules are applied first, followed by the frontend's rules.](https://cbonte.github.io/haproxy-dconv/2.6/configuration.html#4.2-http-response)
+ The `http-response set or delete` stanzas are applied and interpreted in this way by the HAProxy.
+For example: A cluster admin wants to `DENY` response header `X-Frame-Options`. However, a route owner sets  `X-Frame-Options` to `'SAMEORIGIN'` then the haproxy.config will look like the following where
+the response header called `X-Frame-Options` with value 'DENY' set in frontend will override the value `'SAMEORIGIN'` of `X-Frame-Options`
+which was set in the backend section via Route's spec.httpHeaders.actions.
+```conf
+frontend public
+  http-response set-header X-Frame-Options 'DENY' 
+
+frontend fe_sni
+  http-response set-header X-Frame-Options 'DENY' 
+
+frontend fe_no_sni
+  http-response set-header X-Frame-Options 'DENY' 
+
+backend be_secure:openshift-monitoring:alertmanager-main
+  http-response set-header X-Frame-Options 'SAMEORIGIN'
+```
+ However, in case of the HTTP request, HAProxy overrides the values of the header value of the same header name present in the frontend with the value of the
+ header defined in backend. [The rules are evaluated in their declaration order when they are met in a frontend,
+ listen or backend section.](https://cbonte.github.io/haproxy-dconv/2.6/configuration.html#4.2-http-request)
```

Comment:
How is this statement relevant here?

Reply:
I just tried to give more information. However I removed this and added a link of http-request -
 This means that for requests, [the frontend's rules are applied first, followed by the backend's,
and conversely](https://cbonte.github.io/haproxy-dconv/2.6/configuration.html#4.2-http-request),